app/controllers/steps_controller.rb

```diff
@@ -44,4 +46,14 @@ def next_path(params = nil)
   def step_navigation
     @step_navigation ||= StepNavigation.new(self)
   end
+
+  def maybe_skip
```

Comment:
`skip?` has some logic in `ResidentialAddressController`-- are you suggesting I only define this `before_action` there and then include in the base class later?

FWIW, I got the idea for this behavior from the prototype app for michigan benefits. This is how they handled skipping steps.